templates/rhpam73-authoring-ha.yaml

```diff
@@ -1327,18 +1481,152 @@ objects:
-        containerNames:
-        - ${APPLICATION_NAME}-amq
-        from:
-          kind: ImageStreamTag
-          namespace: "${AMQ_IMAGE_STREAM_NAMESPACE}"
-          name: "${AMQ_IMAGE_STREAM_NAME}:${AMQ_IMAGE_STREAM_TAG}"
-      type: ImageChange
-    - type: ConfigChange
+    updateStrategy:
+      type: OnDelete
+    volumeClaimTemplates:
+    - metadata:
+        creationTimestamp: null
+        name: ${APPLICATION_NAME}-amq-pvol
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: ${AMQ_VOLUME_CAPACITY}
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: ${APPLICATION_NAME}-amq-role
+    labels:
+      app: ${APPLICATION_NAME}-amq
+      application: ${APPLICATION_NAME}
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - endpoints
+    verbs:
+    - create
+    - delete
+    - deletecollection
+    - get
+    - list
+    - patch
+    - update
+    - watch
+  - apiGroups:
+    - ""
+    resources:
+    - namespaces
+    verbs:
+    - get
+    - list
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: Role
+  metadata:
+    name: ${APPLICATION_NAME}-amq-scaledown-controller-role
+    labels:
+      app: ${APPLICATION_NAME}-amq
+      application: ${APPLICATION_NAME}
+  rules:
+  - apiGroups:
+    - apps
+    resources:
+    - statefulsets
+    verbs:
+    - get
+    - list
+    - watch
+  - apiGroups:
+    - ""
+    resources:
+    - pods
+    verbs:
+    - get
+    - list
+    - watch
+    - create
+    - delete
+    - patch
+    - update
+  - apiGroups:
+    - ""
+    resources:
+    - persistentvolumeclaims
+    verbs:
+    - get
+    - list
+    - watch
+    - delete
+  - apiGroups:
+    - ""
+    resources:
+    - events
+    verbs:
+    - get
+    - list
+    - watch
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: ${APPLICATION_NAME}-amq-scaledown-controller-openshift-rb
+    labels:
+      app: ${APPLICATION_NAME}-amq
+      application: "${APPLICATION_NAME}"
+  subjects:
+    - kind: ServiceAccount
+      name: ${APPLICATION_NAME}-amq-scaledown-controller-sa
+  roleRef:
+    kind: Role
+    name: ${APPLICATION_NAME}-amq-scaledown-controller-role
+    apiGroup: rbac.authorization.k8s.io
+- apiVersion: rbac.authorization.k8s.io/v1
+  kind: RoleBinding
+  metadata:
+    name: ${APPLICATION_NAME}-amq-rb
+    labels:
+      app: ${APPLICATION_NAME}-amq
+      application: ${APPLICATION_NAME}
+  subjects:
+    - kind: ServiceAccount
+      name: ${APPLICATION_NAME}-amq-sa
+  roleRef:
+    kind: Role
+    name: ${APPLICATION_NAME}-amq-role
+    apiGroup: rbac.authorization.k8s.io
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    labels:
+      application: "${APPLICATION_NAME}"
+    name: ${APPLICATION_NAME}-amq-sa
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    labels:
+      application: "${APPLICATION_NAME}"
+    name: ${APPLICATION_NAME}-amq-scaledown-controller-sa
```

Comment:
Can't we use only one role/role binding as the AMQ templates does?
https://github.com/jboss-container-images/jboss-amq-7-broker-openshift-image/blob/amq-broker-72-dev/templates/amq-broker-72-persistence-clustered.yaml#L22

Reply:
ok, sounds good.